src/cargo/core/shell.rs

```diff
@@ -147,27 +147,61 @@ impl MultiShell {
 }
 
 impl Shell {
-    pub fn create(out: Box<Write + Send>, config: ShellConfig) -> Shell {
+    pub fn create(out: Box<Write + Send>, config: ShellConfig) -> CargoResult<Shell> {
```

Comment:
That sounds like a great idea! I'll be rewriting it in such a manner straight away :)